packages/react-aria-components/src/DateField.tsx

```diff
@@ -110,6 +111,11 @@ export const DateField = /*#__PURE__*/ (forwardRef as forwardRefType)(function D
         slot={props.slot || undefined}
         data-invalid={state.isInvalid || undefined}
         data-disabled={state.isDisabled || undefined} />
+      <HiddenDateInput 
```

Comment:
I wonder if this could be combined with the existing hidden input in DateInput which is used for form submission? Seems like we could potentially only have one, assuming the formats are equivalent. Did you try that?